Saule/JsonApiSerializer.cs

```diff
@@ -72,7 +72,9 @@ public PreprocessResult PreprocessContent(object @object, ApiResource resource,
                     type: resource,
                     baseUrl: requestUri,
                     urlBuilder: UrlPathBuilder,
-                    paginationContext: QueryContext?.Pagination);
+                    paginationContext: QueryContext?.Pagination,
+                    includingContext: QueryContext?.Including,
+                    includedDefault: QueryContext != null ? QueryContext.IncludedDefault : true);

```

Comment:
Really all I meant was that everything related to including resources should live on the including context; IMO there is no need for a separate including-related flag on the query context itself. That way, you always know you have all relevant information when you have the including context.